lib/fastlane/plugin/wpmreleasetoolkit/helper/android_localize_helper.rb

```diff
@@ -41,22 +41,32 @@ def self.merge_string(main_strings, library, string_line)
         return :skipped if skip_string_by_exclusion_list(library, string_name)
       
         # Search for the string in the main file
+        result = :added
         main_strings.xpath('//string').each do | this_string | 
           if (this_string.attr("name") == string_name) then
             # Skip if the string has the content_override tag
             return :skipped if skip_string_by_tag(this_string)
             
-            # Update if needed
-            (if (this_string.content == string_content) then return :found else this_string.content = string_content ; return :updated end)   
+            # If nodes are equivalent, skip
+            return :found if (string_line =~ this_string) 
+            
+            # The string needs an update
+            result = :updated
+            if (this_string.attr("tools:ignore").nil?) 
+              # It can be updated, so remove the current one and move ahead
+              this_string.remove 
+              break 
+            else
+              # It has the tools:ignore flag, so update the content without touching the other attributes
+              this_string.content = string_content
+              return result
+            end 
           end
         end
       
-        # String not found and not in the exclusion list: add to the main file
-        new_element = Nokogiri::XML::Node.new "string", main_strings
-        new_element['name'] = string_name
-        new_element.content = string_content
-        main_strings.xpath('//string').last().add_next_sibling("\n#{" " * 4}#{new_element.to_xml()}")
-        return :added
+        # String not found, or removed because needing update and not in the exclusion list: add to the main file
+        main_strings.xpath('//string').last().add_next_sibling("\n#{" " * 4}#{string_line.to_xml()}")
```

Comment:
```suggestion
        main_strings.xpath('//string').last().add_next_sibling("\n#{" " * 4}#{string_line.to_xml().strip}")
```

As mentioned [in the sister PR](https://github.com/wordpress-mobile/WordPress-Android/pull/12825#pullrequestreview-478274213) I noticed an extra newline when running this new version of the code.

I added a dumb `.strip` to this line and it was gone.